### PR TITLE
Add unit tests

### LINKED ISSUE #3519: Improve unit test coverage of the refactored cli code
**Bug description**
This issue is to improve the unit test coverage in the following directories under cli/core
https://github.com/vmware-tanzu/tanzu-framework/tree/main/cli/core/pkg/discovery
https://github.com/vmware-tanzu/tanzu-framework/tree/main/cli/core/pkg/distribution
https://github.com/vmware-tanzu/tanzu-framework/tree/main/cli/core/pkg/utils

This issue also tracks improvements to unit test coverage in all of https://github.com/vmware-tanzu/tanzu-framework/tree/main/cli/runtime

**Affected product area (please put an X in all that apply)**

- ( ) APIs
- ( ) Addons
- (X) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- ( ) Plugin
- ( ) Security
- (X) Test and Release
- ( ) User Experience
- (X) Developer Experience

**Expected behavior**

**Steps to reproduce the bug**

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

### PR DESCRIPTION
**Pull request by saji-pivotal**
### What this PR does / why we need it
This PR improves the unit test coverage of parts of the cli/core, and cli/runtime modules

### Which issue(s) this PR fixes
Fixes #3519

### Describe testing done for PR
1. Ran `make test` in `cli/core` and `cli/runtime` 
2. Also relying on the unit tests that run as part of the CI checks

### Release note

```release-note

```

### CODE REVIEW

File: cli/runtime/command/deprecation.go [hunk 1]
```diff
@@ -22,16 +22,6 @@ func DeprecateCommandWithAlternative(cmd *cobra.Command, removalVersion, alterna
...
 
-// DeprecateFlag marks the flag as deprecated and hidden with a deprecation message.
-func DeprecateFlag(cmd *cobra.Command, flag, removalVersion string) {
-	f := cmd.Flags().Lookup(flag)
-	if f != nil {
-		msg := fmt.Sprintf("will be removed in version %q.", removalVersion)
-		f.Deprecated = msg
-		f.Hidden = true
-	}
-}
```
Comment (anujc25) [inline cli/runtime/command/deprecation.go:33]: Is there any reason behind remove this function?

Comment (anujc25) [reply cli/runtime/command/deprecation.go:33]: I think it will be helpful in future. So, would be nice to keep this function around.

Comment (saji-pivotal) [reply cli/runtime/command/deprecation.go:33]: That works. I'll re-add it, and add unit tests for it. Thanks for the feedback @anujc25 !

**Review by anujc25 — APPROVED** [review]
LGTM. Thanks for adding this.

**Pull request merged**